pkg/heartbeat/heartbeat.go

```diff
@@ -30,6 +34,7 @@ type Heartbeat struct {
 	LineNumber          *int       `json:"lineno"`
 	Lines               *int       `json:"lines"`
 	LocalFile           string     `json:"-"`
+	LocalFileTemporary  bool       `json:"-"`
```

Comment:
Thre's already a dedicated field called `EntityRaw` which is only populated when a remote file is downloaded. Simply checking if `EntityRaw is not empty` does the job.